qutebrowser/mainwindow/tabwidget.py

```diff
@@ -108,6 +108,17 @@ def set_tab_pinned(self, tab: QWidget,
 
         bar.set_tab_data(idx, 'pinned', pinned)
         tab.data.pinned = pinned
+
+        if config.val.tabs.favicons.show == 'pinned':
```

Comment:
I don't like that this logic is here (especially the `tabs_are_windows` part), because there's an almost exact copy in `tabbedbrowser:_update_favicons`. 

Maybe you could move the code from `_update_favicons` into a method here in tabwidget (which updates a single tabs favicons), and have the tabbedbrowser `_update_favicons` iterate over the tabs calling that method? After that, you should be able to call the tabwidget version of 'update_favicons' here.

Reply:
I smell a conflict with #3613 :wink: